packages/contracts/contracts/ionic/CollateralSwap.sol

```diff
@@ -151,7 +150,7 @@ contract CollateralSwap is Ownable2Step, Exponential, IFlashLoanReceiver {
       if (!transferStatus) {
         revert TransferFailed(address(oldCollateralMarket), borrower, address(this));
       }
-      uint256 redeemResult = oldCollateralMarket.redeemUnderlying(borrowedAmount);
+      uint256 redeemResult = oldCollateralMarket.redeem(amountCTokensToSwap);
```

Comment:
Isn't it better to redeem the exact amount borrowed so that you can ensure you repay it exactly?

Reply:
yes, agreed, especially if `amountUnderlying` is passed to `swapCollateral`